### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by tafaramafemba**
### In this pull request, I have implemented the following: 

1. Created a class Classroom with the following:

- @label instance variable, should be initialized in the constructor.
- Setter and getter for @label (remember about attr_accessor).

2. Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:

- Created the has-many side (a classroom has many students).
- Created the belongs-to side (a student belongs to a classroom).
- Made sure that when adding a student to a classroom it also sets the classroom for the student.
- Made sure that when setting the classroom for a student it also adds it to the classroom's students.

3. Created a class Book with the following:

- @title and @author instance variables should be initialized in the constructor.
- Setters and getters for instance variables (remember about attr_accessor).

4. Created a class Rental with the following:

- @date instance variable should be initialized in the constructor.
- Setter and getter for @date (remember about attr_accessor).

5. Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following  was implemented:

- Created the has-many side of Book and Rental (a book has many rentals).
- Created the belongs-to side of Rental and Book (a rental belongs to a book).
- Created the has-many side of Person and Rental (a person has many rentals).
- Created the belongs-to side of Rental and Person (a rental belongs to a person).
- Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :author, :title, :rentals
+
+  def initialize(author, title)
+    @author = author
+    @title = title
+    @rentals = []
+  end
```
Comment (Mihndim2020) [inline book.rb:8]: - [x] To implement the `has-many` side of the relationship between `Book` and `Rentals`, method for adding rentals should be present in this `Book` class. Something like so;
```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date)
```
Comment (Mihndim2020) [inline rental.rb]: - [x] The constructor of Rental should be modified so Book and Person are set in it to break the `many-to-many ` relationship, something like so: 
```
`def initialize(date, book, person)`

```

File: person.rb [hunk 3]
```diff
@@ -10,6 +10,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (Mihndim2020) [inline person.rb:15]: - [x] To implement the `has-many` side of the relationship between `Person` and `Rentals`, method for adding rentals should be present in this `Person` class. Something like so;
```ruby
  def add_rental(book, date)
    Rental.new(date, self, book)
  end
```

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @tafaramafemba, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean and organized code

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

#### Please also consider:
- [OPTIONAL] updating  your `readme file` to reflect the changes for this milestone as this will make your `readme file` more informative and professional. Checkout this [template](https://github.com/microverseinc/readme-template) and adapt it to your project. Replace all the information in the various sections of the template with information pertaining to your poject.  


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BenMukebo — APPROVED** [review]
`
### Hi @tafaramafemba, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._`

**Pull request merged**